packages/util/src/Environment.ts

```diff
@@ -0,0 +1,20 @@
+/**
+ * Secure method to check if the environment is Node.js.
+ * @returns true if the environment is Node.js, false otherwise.
+ */
+export function isNode(): boolean {
+  // Checking the existence of 'window' variable, or 'process' is not enough,
+  // as those variables can be redefined by inner scopes (by any module/library).
+  // In the 'new Function()' constructor, the execution scope of 'this' is bound
+  // to the global scope and it can be compared to the expected value (the objects
+  // will have the same id if the environment is the expected one).
+  return new Function("try {return this===global}catch(e){ return false}")();
```

Comment:
Is this the best practice way to do it? Could we import some package that does this in the way that is consensus in the Node.js community?

Reply:
I haven't found any "official" best practice. Most people just use `if (typeof window !== 'undefined')` or something similar. But that's extremely weak as the comment above explains.

That function is the most solid solution I've found so far.